### PR TITLE
Update the controller to reconcile for route instead of ArgoCD instance

### PR DESCRIPTION
**Pull request by chetan-rns**
In order to reconcile for the ArgoCD CR instance, the controller expects the ArgoCD CRD to be present beforehand. In the absence of CRD, the controller wouldn't be created. This PR addresses this issue by changing the controller logic to reconcile for the ArgoCD server route. Since Route is a native OpenShift object, we can expect the CRD to be already present.

### CODE REVIEW

File: pkg/controller/argocd/argocd_controller.go [hunk 1]
```diff
@@ -137,39 +110,20 @@ type ReconcileArgoCD struct {
...
-			return reconcile.Result{}, r.deleteConsoleLinkIfPresent(ctx, reqLogger)
-		}
-		// Error reading the object - requeue the request.
-		return reconcile.Result{}, err
-	}
-
-	reqLogger.Info("ArgoCD instance found", "ArgoCD.Namespace:", argocdInstance.Namespace, "ArgoCD.Name", argocdInstance.Name)
-
-	// Set ArgoCD instance as the owner
-	if err := controllerutil.SetControllerReference(argocdInstance, newArgoCDRoute(), r.scheme); err != nil {
```
Comment (bigkevmcd) [inline pkg/controller/argocd/argocd_controller.go:167]: You can remove `newArgoCDRoute` now?

File: pkg/controller/argocd/argocd_controller.go [hunk 2]
```diff
@@ -66,36 +63,16 @@ func newReconciler(mgr manager.Manager) reconcile.Reconciler {
 func add(mgr manager.Manager, r reconcile.Reconciler) error {
 
 	reqLogger := logs.WithValues()
-	reqLogger.Info("Watching ArgoCD")
+	reqLogger.Info("Watching ArgoCD Server Route")
 
-	// Skip controller creation if ArgoCD CRD is not present
-	_, err := mgr.GetRESTMapper().RESTMapping(schema.GroupKind{
-		Group: argocdGroup,
```
Comment (bigkevmcd) [inline pkg/controller/argocd/argocd_controller.go:73]: You don't need either of these constants now?

`argocdKind` or `argocdGroup`

File: pkg/controller/argocd/argocd_controller.go [hunk 3]
```diff
@@ -118,10 +95,6 @@ func filterPredicate(assert func(namespace, name string) bool) predicate.Funcs {
 	}
 }
 
-func assertArgoCD(namespace, name string) bool {
-	return namespace == argocdNS && argocdInstanceName == name
-}
-
 func assertArgoCDRoute(namespace, name string) bool {
```
Comment (bigkevmcd) [inline pkg/controller/argocd/argocd_controller.go:98]: nit: I'd have called this `argoCDRouteFilter`

File: pkg/controller/argocd/argocd_controller.go [hunk 4]
```diff
@@ -137,39 +110,20 @@ type ReconcileArgoCD struct {
...
 }
 
-// Reconcile reads that state of the cluster for a ArgoCD object and makes changes based on the state read
-// and what is in the ArgoCD.Spec
+// Reconcile reads that state of the cluster for a ArgoCD Route object and makes changes based on the state read
+// and what is in the Route.Spec
 // Note:
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *ReconcileArgoCD) Reconcile(request reconcile.Request) (reconcile.Result, error) {
```
Comment (bigkevmcd) [inline pkg/controller/argocd/argocd_controller.go:118]: Should this be renamed, now that it's no longer reconciling ArgoCD objects?

File: pkg/controller/argocd/argocd_controller.go [hunk 5]
```diff
@@ -178,26 +132,29 @@ func (r *ReconcileArgoCD) Reconcile(request reconcile.Request) (reconcile.Result
 		}
 		return reconcile.Result{}, err
 	}
-
 	reqLogger.Info("Route found for argocd-server", "Route.Host", argoCDRoute.Spec.Host)
 
-	consoleLink := newConsoleLink("https://"+argoCDRoute.Spec.Host, "ArgoCD")
+	argocCDRouteURL := fmt.Sprintf("https://%s", argoCDRoute.Spec.Host)
```
Comment (bigkevmcd) [inline pkg/controller/argocd/argocd_controller.go:137]: There's probably nothing we can do here, but it feels weird for us to be calculating the route URL like this.

**Review by bigkevmcd — CHANGES_REQUESTED** [review]

**Comment by bigkevmcd** [comment]
Looks mostly good, a couple of minor cleanups and let's get this in :-)

File: pkg/controller/argocd/argocd_controller.go (hunk 1 above)
Comment (chetan-rns) [reply pkg/controller/argocd/argocd_controller.go:167]: Updated

File: pkg/controller/argocd/argocd_controller.go (hunk 2 above)
Comment (chetan-rns) [reply pkg/controller/argocd/argocd_controller.go:73]: Removed the unused constants

File: pkg/controller/argocd/argocd_controller.go (hunk 3 above)
Comment (chetan-rns) [reply pkg/controller/argocd/argocd_controller.go]: The function name is updated

File: pkg/controller/argocd/argocd_controller.go (hunk 4 above)
Comment (chetan-rns) [reply pkg/controller/argocd/argocd_controller.go]: I've renamed it to `ReconcileArgoCDRoute`. I'm okay with `ReconcileRoute` as well

File: pkg/controller/argocd/argocd_controller.go (hunk 5 above)
Comment (chetan-rns) [reply pkg/controller/argocd/argocd_controller.go:135]: Yes. I wonder why the protocol is not added in the route object. They externally append while displaying the link on the UI

**Review by bigkevmcd — APPROVED** [review]
Looks good.

File: pkg/controller/argocd/argocd_controller.go [hunk 6]
```diff
@@ -118,58 +93,35 @@ func filterPredicate(assert func(namespace, name string) bool) predicate.Funcs {
 	}
 }
 
-func assertArgoCD(namespace, name string) bool {
-	return namespace == argocdNS && argocdInstanceName == name
-}
-
-func assertArgoCDRoute(namespace, name string) bool {
+func argocdRouteFilter(namespace, name string) bool {
```
Comment (wtam2018) [inline pkg/controller/argocd/argocd_controller.go:96]: `argoCDRouteFilter` for funciton name.

Comment (chetan-rns) [reply pkg/controller/argocd/argocd_controller.go]: I've updated the function name to `filterArgoCDRoute`

**Pull request merged**